QuasarHelios2015/src/org/team1540/quasarhelios/Clamp.java

```diff
@@ -11,27 +11,32 @@
 import ccre.ctrl.ExtendedMotor;
 import ccre.ctrl.ExtendedMotorFailureException;
 import ccre.ctrl.FloatMixing;
+import ccre.ctrl.Mixing;
 import ccre.ctrl.PIDControl;
 import ccre.igneous.Igneous;
+import ccre.log.LogLevel;
+import ccre.log.Logger;
 
 public class Clamp {
-    public static FloatStatus height = new FloatStatus();
+    public static FloatStatus heightOrSpeed = new FloatStatus();
+    public static BooleanStatus mode = new BooleanStatus(); // true = speed,
+                                                            // false = height
     public static final BooleanStatus openControl = new BooleanStatus(Igneous.makeSolenoid(3));
 
     public static FloatInputPoll heightReadout;
 
     public static void setup() {
 
-//        FloatInputPoll encoder = Igneous.makeEncoder(10, 11, false);
-        FloatInputPoll encoder = FloatMixing.always(0.0f);
-        //FloatOutput speedControl = Igneous.makeTalonMotor(2, Igneous.MOTOR_REVERSE, 0.1f);
+        FloatInputPoll encoder = Igneous.makeEncoder(10, 11, false);
         ExtendedMotor clampCAN = Igneous.makeCANTalon(1);
-        FloatOutput speedControl;
+        FloatOutput temp = FloatMixing.ignoredFloatOutput;
         try {
-            speedControl = clampCAN.asMode(ExtendedMotor.OutputControlMode.VOLTAGE_FRACTIONAL);
+            temp = clampCAN.asMode(ExtendedMotor.OutputControlMode.VOLTAGE_FRACTIONAL);
         } catch (ExtendedMotorFailureException e) {
-            throw new RuntimeException(e);
+            Logger.severe("Exception thrown when creating clamp motor", e);
         }
```

Comment:
... `temp` isn't that much better, but I guess it suffices if you don't want to change it again. Something like `speedControlTemp` would be much more explanator.